topostats/tracing/dnatracing.py

```diff
@@ -773,13 +894,25 @@ def prep_arrays(image: np.ndarray, labelled_grains_mask: np.ndarray, pad_width:
     # Get bounding boxes for each grain
     region_properties = skimage_measure.regionprops(labelled_grains_mask)
     # Subset image and grains then zip them up
-    cropped_images = [crop_array(image, grain.bbox, pad_width) for grain in region_properties]
-    cropped_images = [np.pad(grain, pad_width=pad_width) for grain in cropped_images]
-    cropped_masks = [crop_array(labelled_grains_mask, grain.bbox, pad_width) for grain in region_properties]
-    cropped_masks = [np.pad(grain, pad_width=pad_width) for grain in cropped_masks]
+    cropped_images = []
+    cropped_positions = []
+    cropped_masks = []
+    for grain in region_properties:
```

Comment:
I think we might be able to use the already existing `.bbox` that each `grain` within `region_properties` already has such that `crop_array()` retains its single responsibility and doesn't return both the `cropped_array` and the `cropped_position`.

We can keep the existing dictionary comprehension as was and add in...

```python
    cropped_positions = [(grain.bbox[0], grain.bbox[1]) for grain in region_properties]
```

As well as not adding extra functionality to `crop_array()` it also saves having to subtract `pad_width` since the original `bbox` aren't padded.

Arguably this could be made a function of its own...

```python
def extract_grain_anchor(region_properties: List(RegionProperties) -> list:
    """Extract the anchor (min_row, min_col) from all labelled regions which is used to align individual traces over the original image.

    Parameters
    ==========
    region_properties: List(RegionProperties)
        A list of region properties returned by skimage.measure.regionprops()

    Returns
    =======
    List(Tuple):
        A list of tuples of the min_row, min_col of each bounding box.
    """"
    return [(grain.bbox[0], grain.bbox[1]) for grain in region_properties]
```

Reply:
See above comment about this 👍